src/Symfony/Component/Config/DefaultConfigCacheFactory.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config;
+
+/**
+ * Default implementation for ConfigCacheFactoryInterface
+ * that will create an instance of ConfigCache.
+ *
+ * @author Matthias Pigulla <mp@webfactory.de>
+ */
+class DefaultConfigCacheFactory implements ConfigCacheFactoryInterface
+{
+    /** @var bool Debug flag passed to the ConfigCache */
+    private $debug;
+
+    /**
+     * @param bool $debug The debug flag to pass to ConfigCache
+     */
+    public function __construct($debug)
+    {
+        $this->debug = $debug;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function cache($file, $callback)
+    {
+        $cache = new ConfigCache($file, $this->debug);
```

Comment:
This comment will probably fall a bit out of scope, but I'd like to comment it either way. 

The `$debug` value is as far as I can see, only used to force a rebuild in dev: [ConfigCache.php#L67-L69](https://github.com/symfony/symfony/blob/eee117285ae84b9aa859814d6d0cef9fd7f05baa/src/Symfony/Component/Config/ConfigCache.php#L67-L69). In my opinion, the code should be as little aware of the dev/prod/debug as possible. 

However, now that there is a factory, this can easily be solved; In debug the set method injecting the factory could actually inject an instance of a `DebugConfigCacheFactory`, which by default generates a `DebugCacheConfig` and just returns `true` on `isFresh()`. That means you most likely won't have to bother with the `debug` value through the code, which simplifies a lot and keeps debug code separated from the normal flow.